packages/metadata/src/providers/githubMetadata.provider.ts

```diff
@@ -0,0 +1,94 @@
+import axios, { AxiosInstance } from "axios";
+import { z } from "zod";
+
+import {
+    Address,
+    ILogger,
+    Token,
+    TokenType,
+    ZKChainMetadata,
+    ZKChainMetadataItem,
+} from "@zkchainhub/shared";
+
+import { IMetadataProvider } from "../interfaces/index.js";
+import { FetchError, InvalidSchema } from "../internal.js";
+
+const TokenSchema = z.object({
+    name: z.string(),
+    symbol: z.string(),
+    coingeckoId: z.string(), // FIXME: on pricing refactor, this should not be part of the token metadata
+    type: z.union([z.literal("erc20"), z.literal("native")]),
+    contractAddress: z
+        .string()
+        .regex(/^0x[a-fA-F0-9]{40}$/)
```

Comment:
Just in case, do you think splitting the validation like this is worth?
```suggestion
        .regex(/^0x[a-fA-F0-9]$/)
        .length(42)
```
You'll get one validation for "is an hex string" and another one for "is the expected length of an address".

Feel free to keep it as it is, it's an extremely minor thing. :saluting_face:

Reply:
I think it'd be less performant to be honest lol. But you'd get one error per validation, which might end up being more informative. I imagine that for this value:
```
0x123456789012345678901234567890123456789
```
With the regex doing everything you'll get something like `regex doesn't match`, if you split the checks you'll get something like `expected length 42, got 41` which might ends up making things easier to debug.

It really depends on the nature of those hex values, if they are 99% of the times 42 chars long I'd not worry about this.

Reply:
did a small refactor, moved validation schemas to own file (i will need them for other provider) and had to make a change to have zod inference work properly

Reply:
mmm i get it, however being a contract address they are always the same length (for now, maybe in the future is like IP that changes from ipv4 to ipv6 xd)